components/match2/wikis/osu/match_group_input_custom.lua

```diff
@@ -0,0 +1,462 @@
+---
+-- @Liquipedia
+-- wiki=osu
+-- page=Module:MatchGroup/Input/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Opponent = require('Module:Opponent')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local TypeUtil = require('Module:TypeUtil')
+local Variables = require('Module:Variables')
+local DateExt = require('Module:Date/Ext')
+local Streams = require('Module:Links/Stream')
+local Match = require('Module:Match')
+
+local MatchGroupInput = Lua.import('Module:MatchGroup/Input', {requireDevIfEnabled = true})
+
+local ALLOWED_VETOES = {'decider', 'pick', 'ban', 'defaultban', 'protect'}
+local ALLOWED_STATUSES = { 'W', 'FF', 'DQ', 'L', 'D' }
+local MAX_NUM_OPPONENTS = 4
+local MAX_NUM_MAPS = 20
+local DEFAULT_BESTOF = 3
+
+local EPOCH_TIME_EXTENDED = '1970-01-01T00:00:00+00:00'
+
+local NOW = os.time(os.date('!*t'))
+local LANG = mw.getContentLanguage()
+local MATCHTIME = tonumber(LANG:formatDate('U', Match.date))
+local THRESHOLD = Match.dateexact and 30800 or 86400
+
+-- containers for process helper functions
+local matchFunctions = {}
+local mapFunctions = {}
+local opponentFunctions = {}
+
+local CustomMatchGroupInput = {}
+
+-- called from Module:MatchGroup
+function CustomMatchGroupInput.processMatch(match)
+	-- Count number of maps, check for empty maps to remove, and automatically count score
+	match = matchFunctions.getBestOf(match)
+	match = matchFunctions.getScoreFromMapWinners(match)
+
+	-- process match
+	Table.mergeInto(
+		match,
+		matchFunctions.readDate(match)
+	)
+	match = matchFunctions.getOpponents(match)
+	match = matchFunctions.getTournamentVars(match)
+	match = matchFunctions.getVodStuff(match)
+	match = matchFunctions.getExtraData(match)
+
+	return match
+end
+
+-- called from Module:Match/Subobjects
+function CustomMatchGroupInput.processMap(map)
+	map = mapFunctions.getExtraData(map)
+	map = mapFunctions.getScoresAndWinner(map)
+	map = mapFunctions.getTournamentVars(map)
```

Comment:
this function expects a second param (match)
need to move this from map processing into match processing to have match available

Reply:
There were this one that exist above this function 
![image](https://github.com/Liquipedia/Lua-Modules/assets/88981446/b5e4f8dd-66d7-41f0-9701-75d71c5f88f4) 

what should I do here? or just change (map) to (map, match)?

Reply:
```suggestion
```
together with additional comments